tests/integration/data_processing/test_data_processor.py

```diff
@@ -0,0 +1,59 @@
+"""Integration tests for the ``DataProcessor``."""
+
+from sdv import load_demo
+from sdv.data_processing import DataProcessor
+from sdv.metadata import SingleTableMetadata
+
+
+def test_data_processor_with_anonymized_columns(tmpdir):
+    """Test the ``DataProcessor``.
+
+    Test that when we update a ``pii`` column this uses ``AnonymizedFaker`` to create
+    a new set of data for that column, while it drops it on the ``transform``.
+
+    Setup:
+        - Load metadata and data.
+        - Parse the old metadata to a new metadata.
+        - Anonymize the field ``occupation``.
+
+    Run:
+        - Create a ``DataProcessor`` with the new metadata.
+        - Fit the ``DataProcessor`` instance.
+        - Transform the data.
+        - Reverse transform the data.
+
+    Side effects:
+        - The column ``occupation`` has been dropped in the ``transform`` process.
+        - The column ``occupation`` has been re-created with new values from the
+          ``AnonymizedFaker`` instance.
+    """
+    # Load metadata and data
+    metadata, data = load_demo('adult', metadata=True)
+    data = data['adult']
+    metadata.to_json(tmpdir / 'adult_old.json')
+    SingleTableMetadata.upgrade_metadata(tmpdir / 'adult_old.json', tmpdir / 'adult_new.json')
+
+    adult_metadata = SingleTableMetadata.load_from_json(tmpdir / 'adult_new.json')
+
+    # Add anonymized field
+    adult_metadata.update_column('occupation', sdtype='job', pii=True)
+
+    # Instance ``DataProcessor``
+    dp = DataProcessor(adult_metadata)
+
+    # Fit
+    dp.fit(data)
+
+    # Transform
+    transformed = dp.transform(data)
+
+    # Reverse Transform
+    reverse_transformed = dp.reverse_transform(transformed)
+
+    # Assert
+    data_unique_occupations = data['occupation'].unique()
+    reverse_transformed_unique_occupations = reverse_transformed['occupation'].unique()
+    substracted = set(reverse_transformed_unique_occupations) - set(data_unique_occupations)
+
+    assert len(reverse_transformed_unique_occupations) == len(substracted)
+    assert 'occupation' not in transformed.columns
```

Comment:
we could also assert that none of the values in the reversed are in the original

Reply:
Yes that is being tested with the `len` basically, if the unique values are `639` and after we rest the original ones they still are `639` (the substracted) it means that no old values are in the new ones 🤔 right?